packages/vim.rb

```diff
@@ -3,61 +3,79 @@
 class Vim < Package
   description 'Vim is a highly configurable text editor built to make creating and changing any kind of text very efficient.'
   homepage 'http://www.vim.org/'
-  version '8.2.1976'
+  @_ver = '8.2.2580'
+  version @_ver
   compatibility 'all'
-  source_url 'https://github.com/vim/vim/archive/v8.2.1976.tar.gz'
-  source_sha256 'd2d8bc28e28e9c5a63be570cdb44be39470621bb57dcbace5abbd86e15690678'
+  source_url 'https://github.com/vim/vim/archive/v8.2.2580.tar.gz'
+  source_sha256 'd0a508ca9726c8ff69bc5f5ab1ebe251c256e01e730f7b36afd03a66c89fcf79'
 
-  binary_url ({
-    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/vim-8.2.1976-chromeos-armv7l.tar.xz',
-     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/vim-8.2.1976-chromeos-armv7l.tar.xz',
-       i686: 'https://dl.bintray.com/chromebrew/chromebrew/vim-8.2.1976-chromeos-i686.tar.xz',
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/vim-8.2.1976-chromeos-x86_64.tar.xz',
+  binary_url({
+    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/vim-8.2.2580-chromeos-armv7l.tar.xz',
+     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/vim-8.2.2580-chromeos-armv7l.tar.xz',
+       i686: 'https://dl.bintray.com/chromebrew/chromebrew/vim-8.2.2580-chromeos-i686.tar.xz',
+     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/vim-8.2.2580-chromeos-x86_64.tar.xz'
   })
-  binary_sha256 ({
-    aarch64: 'd7f92124ba6fc6d6188ab872205fe8e9a866475b6dec9e4a7e88933620284068',
-     armv7l: 'd7f92124ba6fc6d6188ab872205fe8e9a866475b6dec9e4a7e88933620284068',
-       i686: '4439db5c399b4ed5a82128c2a97ca16125be842f0bad4a234566083c239ccd96',
-     x86_64: '7e5660fa1d751e53a6012727fed17e3b4dea23ec3adc1750a1189d9b5908e579',
+  binary_sha256({
+    aarch64: 'fc753eff299bb106dde43f55f7b52d40852f3f2b0651eb869afad92eb1c5132c',
+     armv7l: 'fc753eff299bb106dde43f55f7b52d40852f3f2b0651eb869afad92eb1c5132c',
+       i686: 'a830cb5b36fe8752b90829f5b97dc382868d7dfda4462ed7cb3e37eebd27939a',
+     x86_64: '27f86977743855f0f77f9797a80674eda93977a4309a02bff399364237984104'
   })
 
   depends_on 'vim_runtime'
 
+  def self.preflight
+    abort('Please remove libiconv before building.') if File.exist?("#{CREW_LIB_PREFIX}/libcharset.so")
+  end
+
   def self.preinstall
     gvim = `which gvim 2> /dev/null`.chomp
-    abort "gvim version #{version} already installed.".lightgreen unless "#{gvim}" == ""
+    abort "gvim version #{version} already installed.".lightgreen unless gvim.to_s == ''
```

Comment:
Since this is only checking for the version, you can move this to `def self.preflight`.